ic-log/src/canister/state.rs

```diff
@@ -67,37 +42,44 @@ impl LogState {
     pub fn init(
         &mut self,
         caller: Principal,
-        memory_id: MemoryId,
+        memory: VirtualMemory<DefaultMemoryImpl>,
```

Comment:
The memory impl should be a generic type, check the evm code where we use the same approach

Reply:
Unfortunately, we cannot use generics in canister traits. This would break the mechanism that allows having exported function inside a dependency.